### PR TITLE
[RNMobile] Add Playback Bar Color Settings panel to VideoPress Block Settings

### PR DESCRIPTION
**Pull request by SiobhyB**
Part of an ongoing project to add the VideoPress block to the native apps.

## Proposed changes:

* A native version of the `ColorPanel` component has been introduced and added to the VideoPress block's settings. The main code changes in the component's file can be found in https://github.com/Automattic/jetpack/pull/29567/commits/39c1da60cbeaab7ae2676807db116cbe889fba75.

### Other information:

- [ ] Have you written new tests for your changes, if applicable?
- [ ] Have you checked the E2E test CI results, and verified that your changes do not break them?
- [ ] Have you tested your changes on WordPress.com, if applicable (if so, you'll see a generated comment below with a script to run)?

## Jetpack product discussion

p9ugOq-3mV-p2

## Does this pull request change what data or activity we track or use?

No, it does not.

## Testing instructions:

* Go through the steps to add the VideoPress block on the web, optionally toggle the playback bar colors.
* Open up the app and navigate to the block that was added via the web.
* Tap the settings toggle then tap `Dynamic Color` to open the new settings panel.
* If you'd changed the video's playback bar colors on the web, verify that these are accurately reflected in the panel on native.
* Experiment with the new color settings by either toggling `Dynamic Color` on or selecting colors manually when that toggle is off. 
* Ensure the updated colors are accurately reflected in the video's HTML markup (by switching to the editor's HTML mode). Optionally, also open the block back up on web to verify that the colors are updated in the settings there.

### Screenshots

| Subsheet (toggle on)  | Subsheet (toggle off) |
| ------------- | ------------- |
|  |  |

| Color panel  | Main sheet |
| ------------- | ------------- |
|  |  |

### Screen recording

https://user-images.githubusercontent.com/2998162/226431528-6a329595-1c46-44e8-8873-68dfb1033ed7.mov

### CODE REVIEW

**Review by osullivanchris — DISMISSED** [review]
Reviewing based on the video provided this looks great. I love that you retain the color values that were chosen if the user switches dynamic color on then off again.

One question, should we show some value in the initial 'Dynamic color' row e.g. on/off?

**Review by jhnstn — DISMISSED** [review]
Looks good, nice work!

**Comment by SiobhyB** [comment]
Thanks for the reviews, both!

> One question, should we show some value in the initial 'Dynamic color' row e.g. on/off?

I agree this would be a nice add, does the below match what you were thinking? I went ahead to add it in https://github.com/Automattic/jetpack/pull/29567/commits/be6e534cbd41c1b6104463e1226c2a2b0eac4e84.

https://user-images.githubusercontent.com/2998162/226695229-9042390a-f518-4e7d-b5e7-2fe572ecc366.mov

**Comment by osullivanchris** [comment]
> I agree this would be a nice add, does the below match what you were thinking? I went ahead to add it in https://github.com/Automattic/jetpack/commit/be6e534cbd41c1b6104463e1226c2a2b0eac4e84.

@SiobhyB looks great thanks for adding that 👌 👨‍🍳

File: projects/packages/videopress/src/client/block-editor/blocks/video/components/color-panel/index.native.tsx [hunk 1]
```diff
@@ -0,0 +1,134 @@
...
+					<BottomSheet.NavBar.BackButton onPress={ goBack } />
+					<BottomSheet.NavBar.Heading>
+						{ __( 'Dynamic Color', 'jetpack-videopress-pkg' ) }
+					</BottomSheet.NavBar.Heading>
+				</BottomSheet.NavBar>
+				<PanelBody>
+					<ToggleControl
+						label={ __( 'Dynamic color', 'jetpack-videopress-pkg' ) }
+						help={
+							<>
```
Comment (retrofox) [inline projects/packages/videopress/src/client/block-editor/blocks/video/components/color-panel/index.native.tsx]: Curious if adding the `<>` wrapper here is mandatory

**Review by retrofox — DISMISSED** [review]
Accepted, but I suggest a review from a mobile dev folk :-)

File: projects/packages/videopress/src/client/block-editor/blocks/video/components/color-panel/index.native.tsx (hunk 1 above)
Comment (SiobhyB) [reply projects/packages/videopress/src/client/block-editor/blocks/video/components/color-panel/index.native.tsx]: Oh, thanks for catching that! Removed in https://github.com/Automattic/jetpack/pull/29567/commits/b75ce81cf527dccbdff0b022f9e683c78e5d656a :)

**Review by retrofox — APPROVED** [review]

**Pull request merged**